pkg/scalers/ibmmq_scaler.go

```diff
@@ -0,0 +1,218 @@
+	}
+
+	if val, ok := config.TriggerMetadata["queueName"]; ok {
+		meta.queueName = val
+	} else {
+		return nil, fmt.Errorf("no queue name given")
+	}
+
+	if val, ok := config.TriggerMetadata["queueDepth"]; ok && val != "" {
+		queueDepth, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid targetQueueDepth - must be an integer")
+		} else {
+			meta.targetQueueDepth = queueDepth
+		}
+	} else {
+		fmt.Println("No target depth defined - setting default")
+		meta.targetQueueDepth = defaultTargetQueueDepth
+	}
+
+	if val, ok := config.TriggerMetadata["tls"]; ok {
+		tlsDisabled, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid tls setting: %s", err)
+		}
+		meta.tlsDisabled = tlsDisabled
+	} else {
+		fmt.Println("No tls setting defined - setting default")
+		meta.tlsDisabled = defaultTlsDisabled
+	}
+
+	if val, ok := config.AuthParams["username"]; ok {
+		meta.username = val
+	} else {
+		return nil, fmt.Errorf("no username given")
+	}
+
+	if val, ok := config.AuthParams["password"]; ok {
+		meta.password = val
+	} else {
+		return nil, fmt.Errorf("no password given")
+	}
+
+	return &meta, nil
+}
+
+// IsActive returns true if there are messages to be processed/if we need to scale from zero
+func (s *IBMMQScaler) IsActive(ctx context.Context) (bool, error) {
+	queueDepth, err := s.getQueueDepthViaHttp()
+	if err != nil {
+		return false, fmt.Errorf("error inspecting IBM MQ queue depth: %s", err)
+	}
+	return queueDepth > 0, nil
+}
+
+// getQueueDepthViaHttp returns the depth of the MQ Queue from the Admin endpoint
+func (s *IBMMQScaler) getQueueDepthViaHttp() (int, error) {
+
+	queue := s.metadata.queueName
+	url := s.metadata.host
+
+	var requestJson = []byte(`{"type": "runCommandJSON", "command": "display", "qualifier": "qlocal", "name": "` + queue + `", "responseParameters" : ["CURDEPTH"]}`)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(requestJson))
+	req.Header.Set("ibm-mq-rest-csrf-token", "value")
+	req.Header.Set("Content-Type", "application/json")
+	req.SetBasicAuth(s.metadata.username, s.metadata.password)
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: s.metadata.tlsDisabled},
+	}
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	var response CommandResponse
+	json.Unmarshal(body, &response)
+
+	return response.CommandResponse[0].Parameters.Curdepth, nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *IBMMQScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetQueueLengthQty := resource.NewQuantity(int64(s.metadata.targetQueueDepth), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s", "IBMMQ", s.metadata.queueName)),
```

Comment:
The metric name should be unique, that means that if you specify two or more IBM MQ scalers in one ScaledObject, each scaler should have a different metric name. I wonder if this is enough?
Is viable a usecase, where we have two MQ scalers with the same `queueName` but pointing to the different MQ (server)?